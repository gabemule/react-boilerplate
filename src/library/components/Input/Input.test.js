import React from 'react'
import { render } from 'enzyme'

// components
import { DsInput } from './index'

// config
import InputConfig from './Input.config'

// constants
const InputClass = InputConfig.class
// const InputPropText = "Heading Default Text";

describe('Heading Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(<DsInput />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = render(<DsInput />)
    expect(wrapper).toBeTruthy()
    expect(wrapper.find(InputClass)).toBeTruthy()
  })
})

// describe("Props", () => {
//   InputConfig.props.type.options.forEach(type => {
//     it(`Accept prop type: ${type}`, () => {
//       const wrapper = render(<DsInput />);
//       expect(wrapper.find(`${InputClass}--${type}`)).toBeTruthy();
//     });
//   });
// });
