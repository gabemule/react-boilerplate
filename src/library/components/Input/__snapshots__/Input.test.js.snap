// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <div
    class="ds-input ds-input--medium ds-input--animated  with-custom-props"
  >
    <input
      class="ds-input__input"
      placeholder=""
      type="text"
    />
    <label
      class="ds-input__label"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
